kyo-prelude/shared/src/main/scala/kyo/Parse.scala

```diff
@@ -122,90 +284,202 @@ object Parse:
       */
-    def inOrder[A, B, C, S](p1: A < (Parse & S), p2: B < (Parse & S), p3: C < (Parse & S))(using Frame): (A, B, C) < (Parse & S) =
-        inOrder(Chunk(p1, p2, p3))(using Flat.unsafe.bypass).map { s =>
+    def inOrder[A: Flat, B: Flat, C: Flat, S](
+        parser1: => A < (Parse & S),
+        parser2: => B < (Parse & S),
+        parser3: => C < (Parse & S)
+    )(using
+        Frame
+    ): (A, B, C) < (Parse & S) =
+        inOrder(Chunk(
+            () => parser1,
+            () => parser2,
+            () => parser3
+        ))(using Flat.unsafe.bypass).map { s =>
             (s(0).asInstanceOf[A], s(1).asInstanceOf[B], s(2).asInstanceOf[C])
         }
 
     /** Parses four parsers in sequence and combines their results. This operation is commonly known as "sequence" in other parsing
       * libraries.
       *
-      * @param p1
-      *   First parser to execute
-      * @param p2
-      *   Second parser to execute
-      * @param p3
-      *   Third parser to execute
-      * @param p4
-      *   Fourth parser to execute
       * @return
       *   Tuple containing all four parsed results in order. The parse branch is dropped if any parser fails
       */
-    def inOrder[A, B, C, D, S](p1: A < (Parse & S), p2: B < (Parse & S), p3: C < (Parse & S), p4: D < (Parse & S))(using
-        Frame
+    def inOrder[A: Flat, B: Flat, C: Flat, D: Flat, S](
+        parser1: => A < (Parse & S),
+        parser2: => B < (Parse & S),
+        parser3: => C < (Parse & S),
+        parser4: => D < (Parse & S)
+    )(
+        using Frame
     ): (A, B, C, D) < (Parse & S) =
-        inOrder(Chunk(p1, p2, p3, p4))(using Flat.unsafe.bypass).map { s =>
+        inOrder(Chunk(
+            () => parser1,
+            () => parser2,
+            () => parser3,
+            () => parser4
+        ))(using Flat.unsafe.bypass).map { s =>
             (s(0).asInstanceOf[A], s(1).asInstanceOf[B], s(2).asInstanceOf[C], s(3).asInstanceOf[D])
         }
 
-    /** Selects between multiple successful parses using a selection function. This is useful for implementing custom disambiguation
-      * strategies between multiple valid parses.
+    /** Parses five parsers in sequence and combines their results. This operation is commonly known as "sequence" in other parsing
+      * libraries.
       *
-      * @param seq
-      *   Sequence of parsers to try - all parsers will be evaluated
-      * @param f
-      *   Selection function that receives pairs of successful parses (with their states) and decides which one to keep. Returns Maybe.empty
-      *   to skip both.
       * @return
-      *   The parse result selected by the selection function
+      *   Tuple containing all five parsed results in order. The parse branch is dropped if any parser fails
       */
-    def select[A: Flat, S, S2](seq: (A < (Parse & S))*)(f: ((State, A), (State, A)) => Maybe[(State, A)] < S2)(
```

Comment:
I've decided to remove `Parse.select`, `Parse.shortest`, and `Parse.longest`. I though I was going to need it but I won't. It's also possible to achieve the same behavior via `Parse.attempt`.